akka-http/src/main/scala/akka/http/javadsl/server/JavaScalaTypeEquivalence.scala

```diff
@@ -0,0 +1,63 @@
+package akka.http.javadsl.server
+
+import akka.http.javadsl
+import akka.http.scaladsl
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+import java.util.Optional
+
+object JavaScalaTypeEquivalence {
+  // We define implicit conversions from the javadsl model objects to the scaladsl subclasses,
+  // for all types where there is only 1 valid direct subclass of the javadsl type
+  // (being the scaladsl one)
```

Comment:
Good idea. Let's do it on the next run after I fix the lower-hanging comments. And maybe after the thing _really_ compiles (I totally missed `akka-docs` last time I claimed that...)